Mathlib/RingTheory/Coalgebra/Convolution.lean

```diff
@@ -0,0 +1,137 @@
+/-
+Copyright (c) 2025 Yaël Dillies, Michał Mrugała, Yunzhou Xie. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Michał Mrugała, Yunzhou Xie
+-/
+import Mathlib.LinearAlgebra.TensorProduct.Tower
+import Mathlib.RingTheory.Coalgebra.Basic
+
+/-!
+# Convolution product on linear maps from a coalgebra to an algebra
+
+This file constructs the ring structure on linear maps `C → A` where `C` is a coalgebra and `A` an
+algebra, where multiplication is given by
+```
+         .
+        / \
+f * g = f g
+        \ /
+         .
+```
+-/
+
+suppress_compilation
+
+open Coalgebra TensorProduct
+
+variable {R A B C : Type*} [CommSemiring R]
+
+namespace LinearMap
+
+local notation "η" => Algebra.linearMap R A
+local notation "ε" => counit (R := R) (A := C)
+local notation "μ" => mul' R A
+local notation "δ" => comul
+local infix:70 " ⊗ₘ " => TensorProduct.map
+-- local notation "α" => TensorProduct.assoc _ _ _
+
+section Semiring
+variable [Semiring A] [Semiring B] [AddCommMonoid C] [Algebra R A] [Algebra R B] [Module R C]
+  [Coalgebra R C]
+
+instance instOne : One (C →ₗ[R] A) where one := Algebra.linearMap R A ∘ₗ counit
```

Comment:
Doesn't this create a diamond when C = A?

Reply:
It's deliberately an `abbrev`, the leakage is currently intentional.

Reply:
(I think this was discussed in the Zulip thread)

Reply:
Probably scoping is enough. Do you care about `Ring (A →ₗ[R] A)` for a bialgebra `A`? There are genuinely two possible structures on them. Are they related in anyway and would we need to deal with both of them at the same time?

Reply:
My proposed solution is revive the thread on Zulip. Either scoping or a new type synonym is probably the right approach.

Reply:
I have scoped